pkg/app/validate_app.go

```diff
@@ -134,6 +134,20 @@ func (v *Validator) Validate(request *admissionv1.AdmissionRequest) (bool, error
 		return false, microerror.Mask(err)
 	}
 
+	var currentApp v1alpha1.App
+
+	if request.Operation == admissionv1.Update {
+		if _, _, err := validator.Deserializer.Decode(request.OldObject.Raw, nil, &currentApp); err != nil {
+			return false, microerror.Maskf(parsingFailedError, "unable to parse current app: %#v", err)
```

Comment:
Maybe not super critical: If we have an invalid app which got created for some reason, we will have no way to fix it here.

Reply:
True, since we can't compare the app I think its safer to allow the update but still log the error.

I've updated it. Thx!